### ISSUE TITLE
Gateway level notification handlers.

### ISSUE THREAD

**Issue by tdelenikas** [tdelenikas/smslib#1]
Idea by @derjust

Consider the option to have gateway level handlers (for inbound messages, etc) for notifications. Service level handlers should act as a fallback option.

**Comment by taufikterdidik** [comment]
Yes, I need this feature too, because we no longer need periodic check for inbound message(PHP->database)
